### ISSUE TITLE
python3.8 debugging broken due to AWS Lambda bootstrap changes

### ISSUE THREAD

**Issue by mhart** [awslabs/aws-sam-cli#2069]
### Description

aws-sam-cli makes the assumption that `/var/runtime/bootstrap` on the python3.8 runtime is a python file:

https://github.com/awslabs/aws-sam-cli/blob/89c5ed09f4c791f8f863b1057d3973944b87d80f/samcli/local/docker/lambda_debug_settings.py#L205

Which was true until today – it seems the AWS Lambda runtime team has updated it to be a bash file.

So users currently trying to use aws-sam-cli for debugging the python3.8 runtime will see a syntax error.

`/var/runtime/bootstrap.py` can now be run as main, so the argument could just be updated to that, or aws-sam-cli could use the new `AWS_LAMBDA_EXEC_WRAPPER` env variable that's been introduced on the Amazon Linux 2 runtimes.

**Comment by mhart** [comment]
The runtimes have been rolled back. Closing this – hopefully won't surface again!

**Comment by mhart** [comment]
Reopening as Python 3.8 changes are being rolled out and this will be an issue again.

TLDR; `/var/runtime/bootstrap` is no longer a Python file, as assumed by https://github.com/awslabs/aws-sam-cli/blob/380d64698fa24a898ac34f8f8644c9548dbf6211/samcli/local/docker/lambda_debug_settings.py#L140-L149

Until this is updated, any new python3.8 images will break debugging

**Comment by jfuss** [comment]
Fixed in the [1.0.0 release](https://github.com/awslabs/aws-sam-cli/releases/tag/v1.0.0).

Closing

**Comment by styfle** [comment]
I'm curious, is `AWS_LAMBDA_EXEC_WRAPPER` documented anywhere?

A google search finds only one result: this issue ☺️

**Comment by mhart** [comment]
It's not. It's new. And... that's all I can say about that for now 😆

**Comment by jkim-hyla** [comment]
any chance this is still an issue?  I'm seeing this error now.

**Comment by mhart** [comment]
@jkim-hyla you need to update your SAM CLI version to 1.0.0

**Comment by jkim-hyla** [comment]
> @jkim-hyla you need to update your SAM CLI version to 1.0.0

doh!  that was it!  Thanks for the quick reply!

**Comment by Spothedog1** [comment]
I'm getting this error again and am using SAM CLI version 1.0.0

**Comment by oleksandr-lysenko** [comment]
The same error for me (it was Ok last week).
Is there a way to use an older version of container when run "sam build", "sam local invoke" commands?

**Comment by mhart** [comment]
FWIW @styfle that env variable is finally documented in public 😉

https://docs.aws.amazon.com/lambda/latest/dg/runtimes-modify.html#runtime-wrapper
